### PR TITLE
Add in parameters for Named Values

### PR DESCRIPTION
**Pull request by harmrai**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CNP-1344

### Change description ###

Add in Named Values for supporting parameterisation of policies.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: template/ccdDataStoreAPI.json [hunk 1]
```diff
@@ -29,6 +29,10 @@
       "apiPolicyUrl": {
         "type": "string"
       },
+      "namedValueTag": {
```
Comment (almmechanics) [inline template/ccdDataStoreAPI.json]: `namedValueTag` never used

**Review by almmechanics — CHANGES_REQUESTED** [review]
1. Replace the `apiManagementServiceName`variable with the parameter which has the name for it
1. Let the deployment resource group (i.e.  "[resourceGroup().location]") be used instead of `apiManagementResourceGroup`
1. It may be worth adding in additional metadata/description elements for the ARM parameters to make it easier for handover
1. remove the `secret:false` as the default is false if not provided (https://docs.microsoft.com/en-us/azure/templates/microsoft.apimanagement/2019-01-01/service/properties)

**Comment by almmechanics** [comment]
"contentValue": "[parameters('apiContentValue')]",
            "contentFormat": "swagger-link-json"

should be
"value" and "format"

**Comment by almmechanics** [comment]
Also `authenticationSettings`  and `subscriptionKeyParameterNames` are null by default so they can be lost

**Comment by harmrai** [comment]
> ```
>         "contentValue": "[parameters('apiContentValue')]",
>         "contentFormat": "swagger-link-json"
> ```
> 
> should be
> "value" and "format"

Ahh, look likes these were renamed as part of the newer 2019-01-01 API. 2018-01-01 these were referred to as contentValue and contentFormat.

**Review by almmechanics — COMMENTED** [review]
Changes to VNet config accepted
Definition of named variables accepted

**Review by almmechanics — COMMENTED** [review]
changes accepted

**Review by almmechanics — APPROVED** [review]

**Pull request merged**